packages/palette/src/elements/ResponsiveBox/ResponsiveBox.tsx

```diff
@@ -0,0 +1,51 @@
+import { scale } from "proportional-scale"
+import React from "react"
+import { Box, BoxProps } from "../Box"
+
+/** ResponsiveBoxProps */
+export type ResponsiveBoxProps = Omit<BoxProps, "maxWidth" | "maxHeight"> & {
+  aspectWidth: number
+  aspectHeight: number
+  maxWidth: number
+  maxHeight: number
+}
+
+/** ResponsiveBox */
+export const ResponsiveBox: React.FC<ResponsiveBoxProps> = ({
+  aspectWidth,
+  aspectHeight,
+  maxWidth,
+  maxHeight,
+  children,
+  ...rest
+}) => {
+  const { width, paddingBottom } = scale({
+    width: aspectWidth,
+    height: aspectHeight,
+    maxWidth,
+    maxHeight,
+  })
+
+  return (
+    <Box
+      position="relative"
+      width="100%"
+      style={{ maxWidth: `${width}px` }}
```

Comment:
Yeah, potentially you might see `100%`, in which case the `maxHeight` would be not necessary.

I think it might make sense to alter the type signature here to be either `maxWidth: number, maxHeight: number | maxWidth: '100%'` (and ignore any other potential values)

My main use-case here is for creating a bounding box for images — so generally you've got some upper bound to the image dimensions (the native dimensions at 1x (or have at 2x, etc)). But 100% is definitely a valid use as well.